doc/docs/concepts.md

```diff
@@ -0,0 +1,123 @@
+
+FHIR Data Pipes provides two approaches for flattening the FHIR Resources into virtual or materialized views
+
+1.  SQL queries to generate virtual views (outside the pipeline)
+2.  FHIR ViewDefinition resources to generated materialized views (within the pipeline)
+
+For both of these approaches, a set of **"predefined views”** for common resources are provided. These can be modified or extended.
+
+The currently supported list (as at 1st June, 2024), which can be found in the `docker/config/views' directory are:
+
+```
+Condition, DiagnosticReport, Encounter, Immunization, Location, Medicationrequest, Observation, Organization, Patient, Practitioner, PractitionerRole, Procedure
+```
+
+### SQL virtual views
+These are samples of more complex SQL-on-FHIR queries for defining flat views for common FHIR Resources. These virtual views are applied outside of the pipelines in the downstream SQL query engine. 
+
+The queries, which have .sql suffix can be found here: /docker/config/views (e.g Patient_flat.sql)
+
+An example of a flat view for the Observation Resource is below
+
+```sql
+CREATE OR REPLACE VIEW flat_observation AS
+SELECT O.id AS obs_id, O.subject.PatientId AS patient_id,
+        OCC.`system` AS code_sys, OCC.code,
+        O.value.quantity.value AS val_quantity,
+        OVCC.code AS val_code, OVCC.`system` AS val_sys,
+        O.effective.dateTime AS obs_date
+      FROM Observation AS O LATERAL VIEW OUTER explode(code.coding) AS OCC
+        LATERAL VIEW OUTER explode(O.value.codeableConcept.coding) AS OVCC
+```
+
+### ViewDefinition resource
+The [SQL-on-FHIR-v2 specification](https://build.fhir.org/ig/FHIR/sql-on-fhir-v2/) defines a standards based pattern for defining Views as FHIRPath expressions in a logical structure to specify the column names and values (as unnested items).
+
+A system (pipeline or library) that implements the “View Layer” of the specification provides a View Runner that is able to process these FHIR ViewDefinition Resources over the “Data Layer” (lossless representation of the FHIR data). The output of this are a set of portable, tabular views that can be consumed by the “Analytics Layer” which is any number of tools that can be used to work with the resulting tabular data.
+ 
+FHIR Data Pipes is a reference implementation of the SQL-on-FHIR-v2 specification:
+
+*   The "View Runner" is by default part of the ETL Pipelines and uses the transformed Parquet files as the “Data Layer”. _This can be extracted to be a stand-alone component if required_
+
+*   When enabled as part of the Pipeline configuration, it will apply the ViewDefinition Resources from the /config/views folder and materialize the resulting tables to the PostgresSQL database.
```

Comment:
```suggestion
*   When enabled as part of the Pipeline configuration, it will apply the ViewDefinition resources from the [views folder](https://github.com/google/fhir-data-pipes/tree/master/docker/config/views) and materialize the resulting tables to the configured database (e.g., an instance of PostgresSQL, MySQL, etc.).
```